virtualizarr/writers/icechunk.py

```diff
@@ -96,65 +112,147 @@ def write_variables_to_icechunk_group(
             group=group,
             name=name,
             var=var,
+            append_dim=append_dim,
         )
 
 
-def write_variable_to_icechunk(
-    store: "IcechunkStore",
+def num_chunks(
+    array,
+    axis: int,
+):
+    return array.shape[axis] // array.chunks[axis]
+
+
+def resize_array(
     group: "Group",
     name: str,
     var: Variable,
-) -> None:
-    """Write a single (possibly virtual) variable into an icechunk store"""
-    if isinstance(var.data, ManifestArray):
-        write_virtual_variable_to_icechunk(
-            store=store,
-            group=group,
-            name=name,
-            var=var,
-        )
-    else:
-        raise ValueError(
-            "Cannot write non-virtual variables as virtual variables to Icechunk stores"
-        )
+    append_axis: int,
+) -> Array:
+    existing_array = group[name]
+    # This is the second time we check if the array is an instance of zarr.core.Array
+    # but it's necessary to ensure .shape and .resize are available
+    if not isinstance(existing_array, Array):
+        raise ValueError("Expected existing array to be a zarr.core.Array")
+    new_shape = list(existing_array.shape)
+    new_shape[append_axis] += var.shape[append_axis]
+    existing_array.resize(tuple(new_shape))
+    return existing_array
+
+
+def get_axis(
+    dims: list[str],
+    dim_name: Optional[str],
+) -> int:
+    if dim_name is None:
+        raise ValueError("dim_name must be provided")
+    return dims.index(dim_name)
+
+
+def _check_compatible_arrays(
+    ma: ManifestArray, existing_array: zarr.core.array.Array, append_axis: int
+):
+    manifest_api.check_combinable_zarr_arrays([ma, existing_array])
+    manifest_api.check_same_ndims([ma.ndim, existing_array.ndim])
+    arr_shapes = [ma.shape, existing_array.shape]
+    manifest_api.check_same_shapes_except_on_concat_axis(arr_shapes, append_axis)
```

Comment:
I know I'm being annoying but actually could we move these utility functions to another namespace, e.g. `manifests.utils`? Otherwise we have an "array API" namespace that contains the array API-compatible functions like `concat`/`stack`, but also these random internal utility functions.